### PR TITLE
Fix TotalScoreSnapshotModifiedOn

### PR DESCRIPTION
**Pull request by MTsatsarov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1016

Added method to update totalScoreSnapshot and totalScoreSnapshotModifiedOn

### CODE REVIEW

**Review by gogo4ds — APPROVED** [review]

File: Open Judge System/Services/OJS.Services/Business/ParticipantScores/IParticipantScoresBusinessService.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
     public interface IParticipantScoresBusinessService : IService
     {
-        void RecalculateForParticipantByProblem(int participantId, int problemId);
+        void RecalculateForParticipantByProblem(int participantId, int problemId, bool shouldUpdateTotalScoreSnapshot = true);
```
Comment (gogo4ds) [inline Open Judge System/Services/OJS.Services/Business/ParticipantScores/IParticipantScoresBusinessService.cs:8]: Is this the correct name for the param?

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Open Judge System/Services/OJS.Services/Data/ParticipantScores/IParticipantScoresDataService.cs [hunk 2]
```diff
@@ -21,21 +21,22 @@ public interface IParticipantScoresDataService : IService
...
+        void ResetBySubmission(Submission submission, bool shouldUpdateTotalScoreModfiedOn);
 
         void DeleteAllByProblem(int problemId);
 
         void DeleteForParticipantByProblem(int participantId, int problemId);
 
         void Delete(IEnumerable<ParticipantScore> participantScores);
 
-        void AddBySubmissionByUsernameAndIsOfficial(Submission submission, string userName, Participant participant);
+        void AddBySubmissionByUsernameAndIsOfficial(Submission submission, string userName, Participant participant, bool updateTotalScoreModifiedOn = true);
```
Comment (gogo4ds) [inline Open Judge System/Services/OJS.Services/Data/ParticipantScores/IParticipantScoresDataService.cs:32]: Lets use the same name everywhere

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Open Judge System/Services/OJS.Services/Data/ParticipantScores/ParticipantScoresDataService.cs [hunk 3]
```diff
@@ -125,7 +134,8 @@ public void Delete(IEnumerable<ParticipantScore> participantScoresEnumerable)
         public void AddBySubmissionByUsernameAndIsOfficial(
             Submission submission,
             string userName,
-            Participant participant)
+            Participant participant,
+            bool shouldUpdateTotalScoreModifierOn = true)
```
Comment (gogo4ds) [inline Open Judge System/Services/OJS.Services/Data/ParticipantScores/ParticipantScoresDataService.cs:138]: Here we should always update the date, as it is new score

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Open Judge System/Services/OJS.Services/Business/ParticipantScores/IParticipantScoresBusinessService.cs (hunk 1 above)
Comment (MTsatsarov) [reply Open Judge System/Services/OJS.Services/Business/ParticipantScores/IParticipantScoresBusinessService.cs]: The logic was changed

File: Open Judge System/Services/OJS.Services/Data/ParticipantScores/IParticipantScoresDataService.cs (hunk 2 above)
Comment (MTsatsarov) [reply Open Judge System/Services/OJS.Services/Data/ParticipantScores/IParticipantScoresDataService.cs]: The logic was changed

File: Open Judge System/Services/OJS.Services/Data/ParticipantScores/ParticipantScoresDataService.cs (hunk 3 above)
Comment (MTsatsarov) [reply Open Judge System/Services/OJS.Services/Data/ParticipantScores/ParticipantScoresDataService.cs]: The logic was changed

File: Open Judge System/Services/OJS.Services/Data/ParticipantScores/IParticipantScoresDataService.cs (hunk 2 above)
Comment (MTsatsarov) [reply Open Judge System/Services/OJS.Services/Data/ParticipantScores/IParticipantScoresDataService.cs]: The logic was changed

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**